tools/Replay.hpp

```diff
@@ -256,46 +268,106 @@ struct Stats {
+        _start = std::chrono::steady_clock::now();
+
+        size_t cpuTime = _timer->elapsedTime().count();
+        _timer.reset(new Util::SysStopwatch);
+
+        perfStatsList.push_back(GetStressStats(runMs, phaseAsString));
+        perfStatsList.push_back(GetCPUUSageStats(cpuTime, phaseAsString));
 
-        if (file.tellp() == 0)
+        if(phase == Log::Phase::Edit)
         {
-            file << "Stress (m/s)";
-            file << "\n";
+            std::vector<PerfMetricInfo> statsList;
+
+            perfStatsList.push_back(GetPeakMemoryUsageStats(_peakMemoryUsage, phaseAsString));
+
+            statsList = GetNetworkStats(_bytesRecvd / 1000, _bytesSent / 1000, phaseAsString);
+            for(size_t i = 0; i < statsList.size(); i++)
+            {
+                perfStatsList.push_back(statsList[i]);
+            }
+
+            statsList = _pingLatency.GetLatencyStats("PL", phaseAsString);
+            for(size_t i = 0; i < statsList.size(); i++)
+            {
+                perfStatsList.push_back(statsList[i]);
+            }
+
+            statsList = _tileLatency.GetLatencyStats("TL", phaseAsString);
+            for(size_t i = 0; i < statsList.size(); i++)
+            {
+                perfStatsList.push_back(statsList[i]);
+            }
         }
+    }
+
+    PerfMetricInfo GetStressStats(size_t runMs, std::string testPhase)
+    {
+        PerfMetricInfo stressStats = {testPhase, "Stress run (ms)", runMs};
+        return  stressStats;
+    }
 
-        file << runMs << "\n";
+    PerfMetricInfo GetCPUUSageStats(size_t cpuUsage, std::string testPhase)
+    {
+        PerfMetricInfo cpuStats = {testPhase, "CPU Usage (us)", cpuUsage};
+        return cpuStats;
     }
 
-    void dumpNetworkStatsToCSV(size_t recievedKb, size_t sentKb)
+    std::vector<PerfMetricInfo> GetNetworkStats(size_t recievedKb, size_t sentKb, std::string testPhase)
     {
-        std::ofstream file("Network.csv", std::ios::out | std::ios::app);
+        std::vector<PerfMetricInfo> networkStatsList;
+
+        networkStatsList.push_back(PerfMetricInfo{testPhase, "Bytes recieved (kB)", recievedKb});
+        networkStatsList.push_back(PerfMetricInfo{testPhase, "Bytes sent (kB)", sentKb});
+
+        return networkStatsList;
+    }
+
+    PerfMetricInfo GetPeakMemoryUsageStats(size_t peakMemory, std::string testPhase)
+    {
+        PerfMetricInfo peakMemoryStats = {testPhase, "Peak memory usage (kB)", peakMemory};
+        return peakMemoryStats;
+    }
+
+    void dumpPerfStatsToCSV(std::vector<PerfMetricInfo> perfData)
+    {
+        std::ofstream file("PerformanceMetricsSummary.csv", std::ios::out | std::ios::app);
 
         if(file.tellp() == 0)
         {
-            file << "Bytes recieved (kB)" << ", Bytes sent(kB)";
+            file << "COMMIT HASH" << "," << "DATE" << "," << "TEST" << "," << "PHASE" << "," << "METRIC" << "," << "value";
```

Comment:
Hmm - now all of these legends are ALLCAPS - shouting at me, but the last item is still lower-case "value" - why ? =) I was suggesting that if you have: "Foo, Baa, baz" - something is wrong with baz - instead we got "FOO, BAA, baz" ;-) Can we have: "Foo, Baa, Baz" ?